pyserini/search/_dsearcher.py

```diff
@@ -0,0 +1,198 @@
+from typing import List, Optional
+import faiss
+from dataclasses import dataclass
+import numpy as np
+from tqdm import tqdm
+import tensorflow.compat.v1 as tf
+
+
+@dataclass
+class DenseSearchResult:
+    docid: str
+    score: float
+    raw: Optional[str]
+
+
+@dataclass
+class Document:
+    docid: str
+    raw: Optional[str]
+
+
+class SimpleDenseSearcher:
+    """Simple Searcher for dense representation
+
+    Parameters
+    ----------
+    index_path : str
+        Path to faiss index directory.
+    """
+
+    def __init__(self, index_path: str, doc_path: str):
+        self.index = faiss.read_index(index_path)
+        self.dimension = self.index.d
+        self.num_docs = self.index.ntotal
+        self.idx2id, self.docs = self.load_corpus(doc_path)
+
+    @classmethod
+    def from_prebuilt_index(cls, prebuilt_index_name: str, corpus_name: str = None):
```

Comment:
You need to store the prebuilt download coordinates somewhere, right? For sparse, they're are: https://github.com/castorini/pyserini/blob/master/pyserini/prebuilt_index_info.py

Do you want a separate file, or just add there?

Reply:
just add there i think